Remove unsafe uses of ThreadFront.currentPause.pauseId

hbenl opened the pull request:
In some places (mostly the Elements panel) we still used `ThreadFront.currentPause.pauseId`. The problem is that this property is `null` when the user switched to a new execution point and the new `pauseId` has not yet been received from the backend. That's why `await ThreadFront.getCurrentPauseId()` should be preferred whenever possible.
I think the only uses of `ThreadFront.currentPause.pauseId` that can be considered safe are:
- to check if the user has navigated (or is currently navigating) to a different pause
- if we need to get the `pauseId` synchronously and bail out if it's `null`

This fixes some of the Sentry issues in FE-1056.

bvaughn commented:
Lots of e2e tests failures.

bvaughn commented:
>  That's why await `ThreadFront.getCurrentPauseId()` should be preferred whenever possible.

I'm concerned that having the following two properties/methods disagree is *very unintuitive*:
* `ThreadFront.currentPause.pauseId`
* `ThreadFront.getCurrentPauseId()`

A method with the name "get current thing" should reasonably be expected to return the same value as a property "current thing". Seems like this will be a very easy thing for us to reintroduce.

Can we rename `get currentPause` to `get currentPauseUnsafe` or something that makes it more explicit that it's not the thing we should be using in most cases?

hbenl commented:
> Can we rename get currentPause to get currentPauseUnsafe or something that makes it more explicit that it's not the thing we should be using in most cases?

Theoretically the types should be enough as a warning but in practice it's too common to just append `!` if typescript "annoys" with type errors.
So I've replaced `ThreadFront.currentPause.pauseId` with `ThreadFront.currentPauseIdUnsafe` (I noticed that the other properties of `ThreadFront.currentPause` weren't used anymore so I removed them - I had only added them originally for backward compatibility during the removal of the `Pause` class).
While doing that I stumbled on some dead code which I also removed.

bvaughn commented:
> So I've replaced `ThreadFront.currentPause.pauseId` with `ThreadFront.currentPauseIdUnsafe`

Nice! Love it.

bvaughn reviewed (approved):
Just skimmed it, but this looks like a really nice cleanup 👍🏼
